libpdbg/device.c

```diff
@@ -701,6 +701,73 @@ static void pdbg_targets_init_virtual(struct pdbg_target *node, struct pdbg_targ
 	}
 }
 
+/**
+ * @brief It releases/deletes a target from its target_class list.
+ * 		  Each pdbg_target object (except the virtuals one) has an
+ *        associated target class linked via it's class_link. This
+ * 		  function deletes/removes a target form the list of associated
+ * 		  pdbg_target_class object. For internal use only.
+ * 
+ * @param target The pdbg_target object which will be removed from it's associated
+ * 				 pdbg_target_class list
+ * 
+ * @see pdbg_release_target for more details
+ */
+static void pdbg_del_target_from_target_class_list(struct pdbg_target* target)
+{
+	if (!target)
+		return;
+
+	struct pdbg_target_class* target_class = get_target_class(target);
+	if (target_class)
+		list_del_from(&target_class->targets, &target->class_link);
+}
+
+/**
+ * @brief A function to delete/release an existing pdbg_target object along
+ * 		  with its children from the node list/dev tree. Recursive in nature.
+ * 		  For internal use only.
+ * 
+ * @param target The target node for which the children and then the node itself
+ * 				 will be released/freed
+ * 
+ * @see pdbg_release_dt_root for more details
+ */
+static void pdbg_release_target(struct pdbg_target* target)
+{
+	if (!target)
+		return;
+
+	struct pdbg_target *childTarget, *next = NULL;
+	list_for_each_safe(&target->children, childTarget, next, list)
+		pdbg_release_target(childTarget);
+		
+	if (target->class)
+			pdbg_del_target_from_target_class_list(target);
+
+	struct pdbg_target* parentTarget = target->parent;
+	if (parentTarget)
+		list_del_from(&parentTarget->children, &target->list);
```

Comment:
specify why we are doing this in the method description

Reply:
Added